Topics/System_Design/Authentication_Service_Design.md

```diff
@@ -0,0 +1,85 @@
+# Introduction: Designing an Authentication Service
+
+**Authentication** is the first gateway users pass through when using any digital service to prove their identities. **Authorization** is how we ensure the *right people* are accessing the *right data*.
+A well-designed authentication and authorization service has to strike the correct balance of security and user convenience. This balance shifts depending on what service you're offering; an in-browser casual multiplayer game will have completely different needs for security and user convenience compared to governmental corporate software.
+
+## Table of Contents
+
+1. [Types of Authentication](#auth-types)
+	1. [Multi-Factor Authentication](#mfa)
+	2. [Standardized Protocols](#protocols)
+2. [Sessions and Tokens](#sessions-tokens)
+	1. [Designing Token-Based Authentication](#token-design)
+4. [Design Considerations](#auth-design)
+5. [Common Technologies](#auth-common-tech)
+6. [Further Reading](#further-reading)
+
+## Types of Authentication <a name="auth-types"></a>
+
+We all know about usernames and passwords, or **single-factor authentication**, which grants access to a system when the credentials are correct. However, in most modern systems, this is no longer the case for several reasons:
+1. Users don't keep their passwords secure
+2. Users are vulnerable to phishing schemes
+3. Users easily forget their own credentials
+4. Credentials can be easily cracked
+
+A common theme we'll see is that authentication schemes will try to work around the weakest security link in the system, which are the users themselves. While small improvements can help, like security questions or increasing requirements for password hygiene (minimum N characters, with at least M special characters), modern approaches have better ways of working around this limitation. 
+
+### Multi-Factor Authentication (MFA) <a name="mfa"></a>
+
+Some MFA strategies require users to prove their credentials by demonstrating proof of *physical ownership* of an item. The most common among these is requiring the user to confirm their identity on a **separate device**, like a phone or a laptop; however, a really determined threat actor can bypass this by gaining remote access to the user's devices. 
+
+The best option of MFA strategies is by making the user prove their *inherent characteristics*; that is, things that are unique to them that no one else can ever take from them. **Biometrics** are a good example of this; many smartphones will use fingerprints and facial recognition, as an example.
```

Comment:
I think it would be good to add an `Instead` transition before `The best option` to create a better transition between identifying the problem and providing the solution.

Reply:
OK, sounds good!